rcl/src/rcl/publisher_impl.h

```diff
@@ -25,6 +25,7 @@ struct rcl_publisher_impl_s
   rmw_qos_profile_t actual_qos;
   rcl_context_t * context;
   rmw_publisher_t * rmw_handle;
+  const rosidl_message_type_support_t * type_support;
```

Comment:
Definitely need to note the assumption we are making (in all cases where we are now caching this pointer)
```suggestion
  // Assumption: the type_support used to create the publisher outlives any publisher it is used for
  const rosidl_message_type_support_t * type_support;
```